mmtrack/datasets/otb_dataset.py

```diff
@@ -1,11 +1,106 @@
+import os.path as osp
+import re
+import time
+
+import numpy as np
+from mmdet.datasets import DATASETS
+
+from .base_sot_dataset import BaseSOTDataset
 
 
 @DATASETS.register_module()
-class OTB100Dataset(SOTTestDataset):
-    """OTB100 dataset for the testing of single object tracking.
+class OTB100Dataset(BaseSOTDataset):
+    """OTB100 dataset of single object tracking.
 
-    The dataset doesn't support training mode.
+    The dataset is only used to test.
     """
-    pass
+
+    def __init__(self, ann_file, *args, **kwargs):
+        """Initialization of SOT dataset class.
+
+        Args:
+            ann_file (str): The file contains data information. It will be
+                loaded and parsed in the `self.load_data_infos` function.
+        """
+        self.ann_file = ann_file
+        super().__init__(*args, **kwargs)
+
+    def load_data_infos(self, split='test'):
+        """Load dataset information.
+
+        Args:
+            split (str, optional): Dataset split. Defaults to 'test'.
+
+        Returns:
+            list[dict]: The length of the list is the number of videos. The
+                inner dict is in the following format:
+                    {
+                        'video_path': the video path
+                        'ann_path': the annotation path
+                        'start_frame_id': the starting frame number contained
+                            in the image name
+                        'end_frame_id': the ending frame number contained in
+                            the image name
+                        'framename_template': the template of image name
+                        'init_skip_num': (optional) the number of skipped
+                            frames when initializing tracker
+                    }
+        """
+        print('Loading OTB100 dataset...')
+        start_time = time.time()
+        data_infos = []
+        with open(self.ann_file, 'r') as f:
+            # the first line of annotation file is dataset comment.
+            for line in f.readlines()[1:]:
+                line = line.strip().split(',')
+                if line[0].split('/')[0] == 'Board':
+                    framename_template = '%05d.jpg'
+                else:
+                    framename_template = '%04d.jpg'
+                data_info = dict(
+                    video_path=line[0],
+                    ann_path=line[1],
+                    start_frame_id=int(line[2]),
+                    end_frame_id=int(line[3]),
+                    framename_template=framename_template)
+                # Tracker initializatioin in `Tiger1` video will skip the first
+                # 5 frames. Details can be seen in the official file
+                # `tracker_benchmark_v1.0/initOmit/tiger1.txt`.
+                # Annotation loading will refer to this information.
+                if line[0].split('/')[0] == 'Tiger1':
+                    data_info['init_skip_num'] = 5
```

Comment:
If `tiger1` video skip the first 5 frames, we can directly change the value of `start_frame_id` by plus 5.

Reply:
Video `Tiger1` is special in OTB100.
The start_frame_id is 6 not 1 now. But the first 5 bbox annotations are recorded in the `groundtruth.txt`.
We can't directly use the `start_frame_id` to omit the first 5 frames for all videos.